File: LeetCodeProblemsCMake/Problems/LeetCode1518WaterBottles.cpp
//
// Created by saeed on 10/1/25.
//

#include "LeetCode1518WaterBottles.h"

int LeetCode1518WaterBottles::numWaterBottles(int numBottles, int numExchange) {
    int drankBottleCount{};
    int tempNumBottles{};
    int emptyBottles{};
    while (numBottles + emptyBottles >= numExchange) {
        tempNumBottles = numBottles;
        emptyBottles += numBottles;
        numBottles = emptyBottles / numExchange;
        emptyBottles %= numExchange;
        drankBottleCount += tempNumBottles;
    }
    return drankBottleCount+numBottles;
}

File: LeetCodeProblemsGo/LeetCode1518WaterBottles/LeetCode1518WaterBottles.go
package LeetCode1518WaterBottles

func numWaterBottles(numBottles int, numExchange int) int {
	tempBottlesCount := 0
	emptyBottles := 0
	drankBottles := 0
	for numBottles+emptyBottles >= numExchange {
		tempBottlesCount = numBottles
		emptyBottles += numBottles
		numBottles = emptyBottles / numExchange
		emptyBottles = emptyBottles % numExchange
		drankBottles += tempBottlesCount
	}
	return drankBottles + numBottles
}
